charmcraft/commands/store/__init__.py

```diff
@@ -769,15 +769,27 @@ def fill_parser(self, parser):
     def run(self, parsed_args):
         """Run the command."""
         store = Store(self.config.charmhub)
+        packages_list = store.list_registered_names(include_collaborations=True)
+        package_type: str = "UNKNOWN"
```

Comment:
One last question: in which situation a valid package would not be listed as registered? Should we go ahead with an unknown type, or stop with an error?

Reply:
In no situation that should happened. `store.release` should raise error so the `UNKNOWN` may not able to be used at all. Just put it there in case.

Reply:
So I think we should raise an error (e.g. in an`else` in the `for` below) if the name is not found instead of letting it move forward should the "impossible situation" happen.

Reply:
Another possibility would be to just not retrieve the source and put what we think is the type, and let the store error if the type doesn't match. Then we wouldn't need to download the large list at all.

Reply:
I let it pass because if we added new type, old charmcraft should still able to work. Also the store does not care the type at all, as it is not used in the release API. This is pure for the end user to see what type it is, not for store to verify it, as charmcraft does not know it either.